src/attributes.jl

```diff
@@ -1856,6 +1856,108 @@ function attribute_value_type(::ConstraintConflictStatus)
     return ConflictParticipationStatusCode
 end
 
+"""
+    UserDefinedFunction(name::Symbol, arity::Int) <: AbstractModelAttribute
+
+Set this attribute to register a user-defined function by the name of `name`
+with `arity` arguments.
```

Comment:
Are you allowed to have multiple `UserDefinedFunction`s with the same `name` but different `arity`?

Reply:
Not sure. I've just copied the existing data structures for now. But really, we need only know whether the function is unary so we know the gradient and hessian calls.

Reply:
> Are you allowed to have multiple UserDefinedFunctions with the same name but different arity?

No. But you're potentially allowed a univariate and multivariate operator with the same name.

We use the arity to initialize some of the buffers in the AD system. But we could shift that to resize dynamically in each call site. It only gets used in a couple of places.

Then we could add `UserDefinedFunction(name::Symbol, is_univariate::Bool)`.

Reply:
We could treat that number as the maximum arity of a user-defined function. I think this should mostly just work. Then if you want to register `max(args...)`, you don't need to register `max2`, `max3`, `max4`, etc.

Reply:
I don't think the current behavior needs to change. I was asking more because the documentation wasn't clear.